healthtools/scrapers/base_scraper.py

```diff
@@ -26,35 +28,45 @@ def __init__(self):
             "aws_secret_access_key": AWS["aws_secret_access_key"],
             "region_name": AWS["region_name"]
         })
-        # set up authentication credentials
-        awsauth = AWS4Auth(AWS["aws_access_key_id"], AWS["aws_secret_access_key"], AWS["region_name"], 'es')
-        # client host for aws elastic search service
-        if ES['host']:
-            self.es_client = Elasticsearch(
-                hosts=ES['host'],
-                port=443,
-                http_auth=awsauth,
-                use_ssl=True,
-                verify_certs=True,
-                connection_class=RequestsHttpConnection,
-                serializer=JSONSerializerPython2()
-            )
-        else:
-            self.es_client = Elasticsearch('127.0.0.1')
+        self.small_batch = True if "small_batch" in sys.argv else False
+        try:
+            # client host for aws elastic search service
+            if "aws" in ES["host"]:
+                # set up authentication credentials
+                awsauth = AWS4Auth(AWS["aws_access_key_id"], AWS["aws_secret_access_key"], AWS["region_name"], "es")
+                self.es_client = Elasticsearch(
+                    hosts=ES["host"],
+                    port=ES["port"],
+                    http_auth=awsauth,
+                    use_ssl=True,
+                    verify_certs=True,
+                    connection_class=RequestsHttpConnection,
+                    serializer=JSONSerializerPython2()
+                )
+            else:
+                self.es_client = Elasticsearch("{}:{}".format(ES["host"], ES["port"]))
+        except Exception as err:
+            self.print_error("[{}] - ERROR: Invalid Parameters For ES Client".format(
+                datetime.now().strftime("%Y-%m-%d %H:%M:%S")))
+        # if to save locally create relevant directories
+        if not AWS["s3_bucket"] and not os.path.exists(DATA_DIR):
+            os.mkdir(DATA_DIR)
+            os.mkdir(DATA_DIR + "data")
+            os.mkdir(DATA_DIR + "data/archive")
```

Comment:
@DavidLemayian I tried not to change the s3 keys, but I've got an idea of how to work around that